lib/health_check.rb

```diff
@@ -0,0 +1,91 @@
+class HealthCheck
+  include ChefResource
+
+  REACHABLE = 'reachable'.freeze
+  TIMEOUT = 'timeout'.freeze
+  UNREACHABLE = 'unreachable'.freeze
+  ERRORING = 'erroring'.freeze
+  KEYERROR = 'key error'.freeze
+
+  attr_reader :status, :erchef, :postgres
+
+  def initialize
+    @status = nil
+    @erchef = { status: REACHABLE }
+    @postgres = { status: REACHABLE }
+  end
+
+  #
+  # Do a general health check.
+  #
+  def check
+    erchef_health
+    erchef_authentication
+    postgres_health
+    overall
+  end
+
+  private
+
+  #
+  # Try to ping erchef as an indicator of general health
+  #
+  def erchef_health
+    erchef_health_metric do
+      chef.get_rest 'license'
```

Comment:
Testing locally, hitting the license endpoint on Erchef without authentication is no different than hitting the base api url, so it would probably be better just to ensure we can hit the base url here.